dvc/repo/index.py

```diff
@@ -256,9 +256,7 @@ def partial_imports(
             collect_targets = ensure_list(targets)
 
         pairs = chain.from_iterable(
-            self.stage_collector.collect_granular(
-                target, recursive=recursive, with_deps=True
-            )
+            self.stage_collector.collect_granular(target, recursive=recursive)
```

Comment:
This looks like something that may fix the bug symptoms, but I'm wondering if there is an underlying issue that should be resolved in the `with_deps` behavior?

Reply:
Hmm, my first thought was that for partial imports, we don't care about the deps so removal is the way to go. But when you mention it, it looks like it might be a problem with graph building and DVC being unable to resolve it properly.

Reply:
So does this affect any DVC command that uses `--with-deps` with these type of symlinks in the repo?

Reply:
Yes. Pushed a fix with the latest force-push